presto-spi/src/main/java/com/facebook/presto/spi/type/UnscaledDecimal128Arithmetic.java

```diff
@@ -189,6 +189,9 @@ public static long unscaledDecimalToUnscaledLongUnsafe(Slice decimal)
         return negative ? -lo : lo;
     }
 
+    /*
```

Comment:
I think we should decide on some convention: all operators verify 10^38 or none. We shouldn't have some exceptions or mixed operators. I think checking if value exceeds 10^38 should be explicit since it costs time.